test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -177,6 +181,69 @@ public void initClient() throws IOException {
         assert hasXPack != null;
         assert nodeVersions != null;
     }
+    
+    // Helper class to check warnings in REST responses with sensitivity to versions
```

Comment:
Could you use Javadoc here? I'm just so used to it in this context that anything else is like a cognitive speed bump.